task/rsync/0.1/rsync.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: rsync
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: rsync, files sync
+    tekton.dev/displayName: "rsync"
+spec:
+  description: >-
+        This task can be used to synchronize local and remote files.
+  workspaces:
+    - name: source
+      description: The directory you want to sync.
+    - name: ssh-directory
+      optional: true
+      description: A .ssh directory with private key, known_hosts, config. It will overwrite the authentication by username and password.
+  params:
+    - name: direction
+      type: string
+      default: "local-to-remote"
+    - name: local-path
+      type: string
+      default: "./"
+    - name: remote-ip
+      type: string
+      default: "0.0.0.0"
+    - name: remote-username
+      type: string
+      default: "root"
+    - name: remote-password
+      type: string
+      default: "password"
```

Comment:
shouldn't these be passed via secrets as passing the creds via params is the less secure way?

Reply:
How about using Kubernetes Secret to pass the account password here? I want to support both username/password and ssh auth.

Reply:
yeah Kubernetes Secret will be good :+1:

Reply:
I test this with Kubernetes Secret, but I found the password printed out in the logs. I don't find a good path to hide it. So  I remove the basic auth by username/password.

Maybe, Tekton should provide a way to hide sensitive information.

Reply:
@vinamra28